### PR TITLE
Adaptive card set push button aria-pressed and default push button style options

### PR DESCRIPTION
**Pull request by amal-khalaf**
> Fixes #
#3622  [Screen Reader-Intelligent Support Experience-Feedback]: Voiceover/Talkback are not narrating the state(selected) for the buttons present in the Tell us about your experience chat while navigating using right swipe. 

## Changelog Entry

Fixes [#3622](https://github.com/microsoft/BotFramework-WebChat/issues/3622). [Screen Reader-Intelligent Support Experience-Feedback]: Voiceover/Talkback are not narrating the state(selected) for the buttons present in the Tell us about your experience chat while navigating using right swipe, by [@amal-khalaf](https://github.com/amal-khalaf) in PR [#3710](https://github.com/microsoft/BotFramework-WebChat/pull/3710)

## Description
Added aria-pressed=true and role=button attribute to the selected Adaptive card action button so that the AT announce which action button was pressed by the user

Setting the default background color of the selected adaptive card action button to blue  

-  [X] I have added tests and executed them locally
-  [ ] I have updated `CHANGELOG.md`
-  [ ] I have updated documentation

## Review Checklist

> This section is for contributors to review your work.

-  [ ] Accessibility reviewed (tab order, content readability, alt text, color contrast)
-  [ ] Browser and platform compatibilities reviewed
-  [ ] CSS styles reviewed (minimal rules, no `z-index`)
-  [ ] Documents reviewed (docs, samples, live demo)
-  [ ] Internationalization reviewed (strings, unit formatting)
-  [ ] `package.json` and `package-lock.json` reviewed
-  [ ] Security reviewed (no data URIs, check for nonce leak)
-  [ ] Tests reviewed (coverage, legitimacy)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -43,6 +43,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
...
 -  Fixes [#3683](https://github.com/microsoft/BotFramework-WebChat/issues/3683). Activities should be acknowledged when user scrolls to bottom, by [@compulim](https://github.com/compulim) in PR [#3684](https://github.com/microsoft/BotFramework-WebChat/pull/3684)
 -  Fixes [#3676](https://github.com/microsoft/BotFramework-WebChat/issues/3676). Activities without text should not generate bogus `aria-labelledby`, by [@compulim](https://github.com/compulim) in PR [#3697](https://github.com/microsoft/BotFramework-WebChat/pull/3697)
+-  Fixes [##3622](https://github.com/microsoft/BotFramework-WebChat/issues/3622). [Screen Reader-Intelligent Support Experience-Feedback]: Voiceover/Talkback are not narrating the state(selected) for the buttons present in the Tell us about your experience chat while navigating using right swipe, by [@amal-khalaf](https://github.com/amal-khalaf) in PR [#3710](https://github.com/microsoft/BotFramework-WebChat/pull/3710)
```
Comment (corinagum) [inline CHANGELOG.md:46]:
```suggestion
-  Fixes [##3622](https://github.com/microsoft/BotFramework-WebChat/issues/3622). Apply pushed button style options and `aria-pressed` on Adaptive Cards selected buttons, by [@amal-khalaf](https://github.com/amal-khalaf) in PR [#3710](https://github.com/microsoft/BotFramework-WebChat/pull/3710)
```

**Review by corinagum — APPROVED** [review]
LGTM! Just one comment

**Comment by corinagum** [comment]
@amal-khalaf  FYI it looks like this PR is missing a few snapshots, could you verify?

File: packages/bundle/src/adaptiveCards/Attachment/AdaptiveCardRenderer.js [hunk 2]
```diff
@@ -50,6 +50,22 @@ function isPlainObject(obj) {
   return Object.getPrototypeOf(obj) === Object.prototype;
 }
 
+/**
+ *
+ * @param {* The current active element} element
+ * @param {* The name of attribute to be added to the element properties} qualifiedName
+ * @param {* the target value to set for the attribute} nextValue
+ */
+function setAttribute(element, qualifiedName, nextValue) {
```
Comment (compulim) [inline packages/bundle/src/adaptiveCards/Attachment/AdaptiveCardRenderer.js:59]: This seems copying code from the function `setAttributeWithUndo`, can we refactor them?

File: CHANGELOG.md [hunk 3]
```diff
@@ -53,6 +53,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
    -  Although it no locker lock up microphone, clicking on the microphone button has no effect because Direct Line Speech does not support aborting speech recognition
 -  Fixes [#3421](https://github.com/microsoft/BotFramework-WebChat/issues/3421). With Direct Line Speech, after not able to recognize any speech, it should no longer stop working, by [@compulim](https://github.com/compulim) in PR [#3694](https://github.com/microsoft/BotFramework-WebChat/pull/3694)
 -  Fixes [#3616](https://github.com/microsoft/BotFramework-WebChat/issues/3616). [Accessibility documentation] Update activity timestamp grouping to match visual UI, by [@amal-khalaf](https://github.com/amal-khalaf) in PR [#3708](https://github.com/microsoft/BotFramework-WebChat/pull/3708)
+-  Fixes [##3622](https://github.com/microsoft/BotFramework-WebChat/issues/3622). Apply pushed button style options and `aria-pressed` on Adaptive Cards selected buttons, by [@amal-khalaf](https://github.com/amal-khalaf) in PR [#3710](https://github.com/microsoft/BotFramework-WebChat/pull/3710)
```
Comment (compulim) [inline CHANGELOG.md:56]:
```suggestion
-  Fixes [#3622](https://github.com/microsoft/BotFramework-WebChat/issues/3622). Apply pushed button style options and `aria-pressed` on Adaptive Cards selected buttons, by [@amal-khalaf](https://github.com/amal-khalaf) in PR [#3710](https://github.com/microsoft/BotFramework-WebChat/pull/3710)
```

File: packages/bundle/src/adaptiveCards/Styles/StyleSet/AdaptiveCardRenderer.js [hunk 4]
```diff
@@ -1,4 +1,10 @@
-export default function ({ accent, paddingRegular, primaryFont }) {
+export default function ({
+  cardPushButtonBackGroundColor,
```
Comment (compulim) [inline packages/bundle/src/adaptiveCards/Styles/StyleSet/AdaptiveCardRenderer.js:2]:
```suggestion
  cardPushButtonBackgroundColor,
```

File: packages/component/src/Styles/defaultStyleOptions.js [hunk 5]
```diff
@@ -200,7 +200,11 @@ const DEFAULT_OPTIONS = {
   emojiSet: true, // true || false || { ':)' : '😊'}
 
   // Internal
-  internalLiveRegionFadeAfter: 1000
+  internalLiveRegionFadeAfter: 1000,
+
+  //AdaptiveCard push button
+  cardPushButtonBackGroundColor: '#0063B1',
```
Comment (compulim) [inline packages/component/src/Styles/defaultStyleOptions.js:206]:
```suggestion
  cardPushButtonBackgroundColor: '#0063B1',
```

File: packages/bundle/src/adaptiveCards/Styles/StyleSet/AdaptiveCardRenderer.js [hunk 6]
```diff
@@ -54,6 +60,12 @@ export default function ({ accent, paddingRegular, primaryFont }) {
         color: '#6D6D6D'
       },
 
+      '& .ac-pushButton[aria-pressed="true"]': {
+        backgroundColor: cardPushButtonBackGroundColor,
```
Comment (compulim) [inline packages/bundle/src/adaptiveCards/Styles/StyleSet/AdaptiveCardRenderer.js:64]:
```suggestion
        backgroundColor: cardPushButtonBackgroundColor,
```

**Review by compulim — CHANGES_REQUESTED** [review]

File: packages/bundle/src/adaptiveCards/Attachment/AdaptiveCardRenderer.js [hunk 7]
```diff
@@ -434,6 +450,9 @@ const AdaptiveCardRenderer = ({ actionPerformedClassName, adaptiveCard, disabled
   }, [adaptiveCard]);
 
   useEffect(() => {
+    setAttribute(adaptiveCardElementRef.current, 'aria-pressed', 'true') &&
```
Comment (compulim) [inline packages/bundle/src/adaptiveCards/Attachment/AdaptiveCardRenderer.js:453]: Are we intentionally ignoring the undo function?

File: packages/bundle/src/adaptiveCards/Attachment/AdaptiveCardRenderer.js [hunk 8]
```diff
@@ -434,6 +450,9 @@ const AdaptiveCardRenderer = ({ actionPerformedClassName, adaptiveCard, disabled
...
+      setAttribute(adaptiveCardElementRef.current, 'role', 'button');
```
Comment (compulim) [inline packages/bundle/src/adaptiveCards/Attachment/AdaptiveCardRenderer.js:454]: If the `aria-pressed` already has value of `true`, it won't set `role` to `button`. Is this intentional?

**Review by compulim — CHANGES_REQUESTED** [review]

File: packages/bundle/src/adaptiveCards/Attachment/AdaptiveCardRenderer.js (hunk 8 above)
Comment (amal-khalaf) [reply packages/bundle/src/adaptiveCards/Attachment/AdaptiveCardRenderer.js]: fixed

File: packages/bundle/src/adaptiveCards/Attachment/AdaptiveCardRenderer.js (hunk 7 above)
Comment (amal-khalaf) [reply packages/bundle/src/adaptiveCards/Attachment/AdaptiveCardRenderer.js]: fixed

File: packages/bundle/src/adaptiveCards/Attachment/AdaptiveCardRenderer.js (hunk 2 above)
Comment (amal-khalaf) [reply packages/bundle/src/adaptiveCards/Attachment/AdaptiveCardRenderer.js]: removed and will use the original setAttributewithundo

File: packages/bundle/src/adaptiveCards/Styles/StyleSet/AdaptiveCardRenderer.js (hunk 4 above)
Comment (compulim) [reply packages/bundle/src/adaptiveCards/Styles/StyleSet/AdaptiveCardRenderer.js:2]: This is not fixed yet. There is a typo, "BackGround" -> "Background".

File: packages/bundle/src/adaptiveCards/Styles/StyleSet/AdaptiveCardRenderer.js (hunk 6 above)
Comment (compulim) [reply packages/bundle/src/adaptiveCards/Styles/StyleSet/AdaptiveCardRenderer.js:64]: This is not fixed yet.

File: packages/component/src/Styles/defaultStyleOptions.js (hunk 5 above)
Comment (compulim) [reply packages/component/src/Styles/defaultStyleOptions.js:220]: This is not fixed yet.

**Review by compulim — APPROVED** [review]
Please fix the typos. Other than that, it LGTM.

**Pull request merged**